test/intern/mixins/ColumnSet.js

```diff
@@ -94,4 +128,77 @@ define([
 			grid.focus(grid.cell(1, '1-0-1'));
 		});
 	});
+
+	test.suite('ColumnSet + Tree mixins', function () {
+		test.before(function () {
+			var TreeColumnSetGrid = declare([ OnDemandGrid, ColumnSet, Tree ]);
+			var data = [];
+			var store;
+
+			for (i = 0; i < 5; i++) {
+				var parentId = '' + i;
+				data.push({
+					id: parentId,
+					value: 'Root ' + i,
+					name: 'Root ' + i,
+					type: 'country'
+				});
+				for (k = 0; k < 100; k++) {
+					data.push({
+						id: i + ':' + k,
+						parent: parentId,
+						value: 'Child ' + k,
+						name: 'Child ' + i,
+						type: 'city',
+						hasChildren: false
+					});
+				}
+			}
+
+			store = createHierarchicalStore({
+				data: data
+			});
+
+			grid = new TreeColumnSetGrid({
+				collection: store,
+				enableTreeTransitions: false,
+				columnSets: [[
+					[ {renderExpando: true, label:"Name", field:"name", sortable: false} ],
+					[ {label:"Type", field:"type", sortable: false} ]
+				]]
+			});
+
+			document.body.appendChild(grid.domNode);
+			grid.startup();
+		});
+
+		test.afterEach(function () {
+			for (var i = handles.length; i--;) {
+				handles[i].remove();
+			}
+		});
+
+		test.after(function () {
+			grid.destroy();
+		});
+
+		test.test('expand after scroll-x should reset scrollLeft', function () {
+			var column = query('#' + grid.id + '-row-0 .dgrid-column-set')[0];
+			grid.styleColumn('0-0-0', 'width: 10000px; overflow-x: scroll;');
+
+			return expand(0)
+				.then(function () {
+					collapse(0)
+						.then(function () {
+							scrollDistance(column, 'x', 300)
+								.then(function () {
```

Comment:
This should be rewritten to avoid this sort of cascade.

```js
    return expand(0).then(function () {
        return collapse(0);
    }).then(...);
```

etc.